### PR TITLE
Simplified distro.String

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
It makes no sense to print so much information in the String method. For reference, we used to print all of this:

```
name: Ubuntu
guid: '{bae22193-b9f2-4ca1-9194-f99dc5d81730}}'
configuration:
  - Version: 2
  - DefaultUID: 0
  - InteropEnabled: true
  - PathAppended: true
  - DriveMountingEnabled: true
  - undocumentedWSLVersion: 2
  - DefaultEnvironmentVariables:
    - HOSTTYPE: x86_64
    - LANG: en_US.UTF-8
    - PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
    - TERM: xterm-256color
```

If this was needed, yaml.Unmarshal would be the way to go.

If something goes wrong, the error was printed as well:
```
name: wsltesting_TestDistroString_432343647698211282_\x00_invalid_name
guid: distro is not registered
configuration: |
  could not access configuration for wsltesting_TestDistroString_432343647698211282_\x00_invalid_name: WslGetDistributionConfiguration: could not convert distro name to UTF16
```

This makes the mocks more complicated to implement because the error message must be the same for tests to pass.

### CODE REVIEW

File: distro.go [hunk 1]
```diff
@@ -201,59 +200,11 @@ func (d Distro) GetConfiguration() (c Configuration, err error) {
...
-	return fmt.Sprintf(`configuration:
-  - Version: %d
-  - DefaultUID: %d
-  - InteropEnabled: %t
-  - PathAppended: %t
-  - DriveMountingEnabled: %t
-  - undocumentedWSLVersion: %d
-  - DefaultEnvironmentVariables:%s`, c.Version, c.DefaultUID, c.InteropEnabled, c.PathAppended,
-		c.DriveMountingEnabled, c.UndocumentedWSLVersion, fmtEnvs)
+	return fmt.Sprintf("distro{Name: %q, GUID: %q}", d.Name(), guid)
```
Comment (didrocks) [inline distro.go:207]: I think this may be a little bit too goish for a `String()`, this may be better to be more plain english.

**Review by didrocks — CHANGES_REQUESTED** [review]
I thin we should rework the string, but as we discussed, on the principle: +7/-77 \o/

File: distro.go (hunk 1 above)
Comment (EduardGomezEscandell) [reply distro.go:207]: Yeah I'm suffering a terrbile lack of inspiration 😅 

Maybe something like `WSL distro "${Name}" with GUID "${GUID}"` maybe?

I think we can even skip the GUID.

Comment (EduardGomezEscandell) [reply distro.go:207]: Alright 👍

**Review by didrocks — APPROVED** [review]
excellent, thanks!

**Comment by EduardGomezEscandell** [comment]
Huh, tests failed because a distro was stuck in this state even after rebooting the machine

```
wsltesting_TestTerminate_8683048680539705577    Uninstalling    2
```

I had to manually remote into the machine and `wsl --unregister ...` it. Should be solved now.

**Pull request merged**